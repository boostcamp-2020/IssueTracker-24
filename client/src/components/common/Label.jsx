import React from 'react';
import styled from 'styled-components';

const LabelWrapper = styled.button`
  background-color: ${(props) => props.color};
  border: none;
  height: 25px;
  cursor: pointer;
  border-radius: 10%;
  font-weight: bold;
  margin-left: 5px;
  outline: 0;
`;

const Label = ({ label }) => {
  const { id, title, color } = label;
  return (
    <>
      <LabelWrapper key={'label' + id} color={color}>
        <div>{label.title}</div>
      </LabelWrapper>
    </>
  );
};

export default Label;
