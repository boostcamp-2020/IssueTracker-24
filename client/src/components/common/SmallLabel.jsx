import React from 'react';
import styled from 'styled-components';

const Label = styled.div`
  background-color: ${(props) => props.color};
  width: 20px;
  height: 20px;
  border-radius: 15px;
  margin-right: 10px;
`;

const SmallLabel = ({ color }) => {
  return <Label color={color}></Label>;
};

export default SmallLabel;
