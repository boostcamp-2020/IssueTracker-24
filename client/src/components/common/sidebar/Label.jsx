import React, {useState} from 'react';
import styled from 'styled-components'; 
import SmallLabel from '../SmallLabel';
import svg from '../../../utils/svg';
const LabelWrapper = styled.div`
  border-bottom: 1px solid #eaecef;
  cursor:pointer;
  &:hover{
    background-color:#0366D6;
    color:#fff;
 }
 .display-visible {
  visibility: visible;
}
.display-hidden {
  visibility: hidden;
}
`;
const LabelOptionContent = styled.div`
  flex-basis: 80%;
  display: flex;
  justify-content:space-between;
  width: 100%;
`;
const CheckWrapper = styled.div`
  text-align: center;
  flex-basis: 10%;
`;
const LabelOptionTitle = styled.div``;
const LabelOptionDescription = styled.div`
  display:flex;
  justify-content:center;
`;
const CancelWrapper = styled.div`
`;
const LabelHeader = styled.div`
  display:flex;
`;
const Label = ({color, title, description}) =>{
  const [checked, setCheck] = useState(false);
  const checkDisplay = checked ? 'display-visible' : 'display-hidden';
  const handleOnClick = () =>{
    setCheck(!checked);
  }
  
  return (
        <LabelWrapper onClick={handleOnClick}>
          <LabelOptionContent>
            <CheckWrapper className={checkDisplay}>{svg.checkIcon}</CheckWrapper>
            <LabelHeader>
              <SmallLabel color={color} size={14} marginTop={2} ></SmallLabel>
              <LabelOptionTitle>{title}</LabelOptionTitle>
            </LabelHeader>  
             <CancelWrapper className={checkDisplay}>{svg.cancelButton}</CancelWrapper>
          </LabelOptionContent> 
          <LabelOptionDescription>{description}</LabelOptionDescription>
       
        </LabelWrapper>
  );
}
export default Label;