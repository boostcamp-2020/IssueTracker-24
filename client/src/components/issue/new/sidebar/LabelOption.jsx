import React from 'react';
import styled from 'styled-components';
import SmallLabel from '../../../common/SmallLabel';

const LabelOptionWrapper = styled.div`
  color: #586069;
  font-size: 12px;
  padding: 5px 0px 5px 15px;
  &:hover {
    background-color: #0366d6;
    color: white;
  }
  .header {
    display: flex;
  }
`;

const LabelOption = ({ data }) => {
  console.log(SmallLabel);
  return (
    <LabelOptionWrapper color={data.color}>
      <div className="header">
        <SmallLabel color={data.color} />
        <div className="title">{data.title}</div>
      </div>
      <div className="description">{data.description}</div>
    </LabelOptionWrapper>
  );
};

export default LabelOption;
