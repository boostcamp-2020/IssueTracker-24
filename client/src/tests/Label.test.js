import React from 'react';
import { render } from '@testing-library/react';
import LabelOption from '../components/issue/new/sidebar/LabelOption';
import Label from '../components/common/Label';
import SmallLabel from '../components/common/SmallLabel';

describe('<LabelOption/>', () => {
    const data={
      color:'#ffffff',
      title:'bug',
      description:'버그 라벨'
    }
    it('matches snapshot', () => {
      const utils = render(<LabelOption data={data}/>);
      expect(utils.container).toMatchSnapshot();
    });
});


describe('<Label/>', () => {
    const label={
      id:'4',
      color:'#ffffff',
      title:'bug',
    }
    it('matches snapshot', () => {
      const utils = render(<Label label={label}/>);
      expect(utils.container).toMatchSnapshot();
    });
});

describe('<SmallLabel/>', () =>{
    const color="#ffffff";
    it('matches snapshot', () => {
      const utils = render(<SmallLabel color={color}/>);
      expect(utils.container).toMatchSnapshot();
    });
});