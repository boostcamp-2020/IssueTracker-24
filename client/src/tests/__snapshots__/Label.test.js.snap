// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label/> matches snapshot 1`] = `
<div>
  <button
    class="sc-dlnjPT JvOQF"
    color="#ffffff"
  >
    <div>
      bug
    </div>
  </button>
</div>
`;

exports[`<LabelOption/> matches snapshot 1`] = `
<div>
  <div
    class="sc-gtssRu kbUSdc"
    color="#ffffff"
  >
    <div
      class="header"
    >
      <div
        class="sc-bdnylx dlRZXl"
        color="#ffffff"
      />
      <div
        class="title"
      >
        bug
      </div>
    </div>
    <div
      class="description"
    >
      버그 라벨
    </div>
  </div>
</div>
`;

exports[`<SmallLabel/> matches snapshot 1`] = `
<div>
  <div
    class="sc-bdnylx dlRZXl"
    color="#ffffff"
  />
</div>
`;
